packages/starksheet-webapp/src/provider/starknetProvider.ts

```diff
@@ -90,11 +109,11 @@ export class StarknetProvider implements ChainProvider {
     ];
   }
 
-  async callContract(call: {
-    contractAddress: string;
-    entrypoint: string;
-    calldata?: any;
-  }): Promise<any> {
-    return this.rpcProvider.callContract(call, "latest");
+  /**
+   * @inheritDoc
+   */
+  async callContract<T>(call: ContractCall): Promise<T> {
+    const response = await this.rpcProvider.callContract(call, "latest");
+    return response.result[0] as T;
```

Comment:
We never query tuples (nor handle them properly in the frontend),  not sure either about chain support for returning tuples, it seemed cleaner to me to make the `result[0]` here rather than on every `callContract()` call. ethers.js has  another way of representing tuple returns so this seems more generic as well

Reply:
actually I think that starknet returns tuple for a single call, this tuple being possibly only one single item long